src/zenml/cli/base.py

```diff
@@ -64,18 +68,37 @@ def init(path: Optional[Path]) -> None:
 
 
 @cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
+@click.option("--yes", "-y", is_flag=True, default=False)
 def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+    """Delete all ZenML metadata and artifacts.
+
+    This is a destructive operation, primarily intended for use in development.
 
     Args:
       yes: bool:  (Default value = False)
     """
     if not yes:
-        _ = confirmation(
-            "This will completely delete all pipelines, their associated "
+        confirm = confirmation(
+            "DANGER: This will completely delete all pipelines, their associated "
             "artifacts and metadata ever created in this ZenML repository. "
             "Are you sure you want to proceed?"
         )
 
-    error("Not implemented for this version")
+    if yes or confirm:
+        local_zen_repo_config = Path.cwd() / REPOSITORY_DIRECTORY_NAME
+        global_zen_config = Path(get_global_config_directory())
+        if fileio.exists(str(local_zen_repo_config)):
+            fileio.rmtree(str(local_zen_repo_config))
+            declare(f"Deleted local ZenML config from {local_zen_repo_config}.")
+        if fileio.exists(str(global_zen_config)):
+            config_yaml_path = global_zen_config / "config.yaml"
+            config_yaml_data = yaml_utils.read_yaml(str(config_yaml_path))
+            fileio.rmtree(str(global_zen_config))
+            declare(f"Deleted global ZenML config from {global_zen_config}.")
+
+        fileio.makedirs(str(global_zen_config))
+        yaml_utils.write_yaml(str(config_yaml_path), config_yaml_data)
```

Comment:
The way this works it wont actually delete the profiles, but still leave a reference to them in the `config.yaml`. Here is an example config file

```
activated_profile: zenbytes
analytics_opt_in: false
profiles:
  default:
    active_stack: default
    name: default
    store_type: local
    store_url: file:///home/hamza/.config/zenml/profiles/default
  legacy-repository-2dc8050f:
    active_stack: aws_kubeflow_stack
    name: legacy-repository-2dc8050f
    store_type: local
    store_url: file:///home/hamza/workspace/zenml_io/zenml/.zen
  mlflow-as-stack:
    active_stack: default
    name: mlflow-as-stack
    store_type: local
    store_url: file:///home/hamza/.config/zenml/profiles/mlflow-as-stack
  wandb-as-stack:
    active_stack: default
    name: wandb-as-stack
    store_type: local
    store_url: file:///home/hamza/.config/zenml/profiles/wandb-as-stack
  zenbytes:
    active_stack: aws_kubeflow_stack
    name: zenbytes
    store_type: local
    store_url: file:///home/hamza/.config/zenml/profiles/zenbytes
user_id: 1f51b5b2-bac9-4a5b-9433-4b04e8eff5b6
version: 0.7.2
```

If you restore the `profiles` key they will now point to folders that dont exist :-) 

The cleanest way to do this is to use the logic of what is happening in `delete_profile` here (https://github.com/zenml-io/zenml/blob/aac824bd2b1733dec20ef7c70af1013bdc549cc9/src/zenml/cli/config.py#L209) . You can simply loop over all the profiles and do `cfg.delete_profile(name)` to make sure the deletion is done through the same canonical function

Reply:
The only thing we need to preserve is the `user_id`, if I'm not mistaken. What is the point in using the `cfg` to delete things when I can just reinstantiate and switch out the `user_id`. Is this going to cause telemetry issues? I'm pushing a solution that'd do it the way I thought made sense.